Source/cmGlobalFastbuildGenerator.cxx

```diff
@@ -262,6 +262,29 @@ class cmGlobalFastbuildGenerator::Detail::FileContext
 class cmGlobalFastbuildGenerator::Detail::Detection
 {
 public:
+
+	static std::string GetLastFolderName(const std::string& string)
+	{
+		return string.substr(string.rfind('/'));
+	}
+
+	static bool HasParentExcludedFromAll( cmTarget* target )
+	{
+		bool excluded = target->GetPropertyAsBool("EXCLUDE_FROM_ALL");
+		cmLocalGenerator* lg = target->GetMakefile()->GetLocalGenerator();
+		while(lg->GetParent() != 0)
+		{
+			excluded = lg->GetMakefile()->GetPropertyAsBool("EXCLUDE_FROM_ALL");
```

Comment:
I think this line has overwritten the property on the object itself. So the initial value of excluded is ignored...

Either change this to an OR, or change the loop conditions  to exit if excluded is true. That would mean you can remove the test and break in the middle of the loop too